lblod/vendor-login-service
# Package and image bumps

## Ticket Number

DL-5668

## Description

In order to allow access to the vendor API, the `vendor-login-service` must be integrated in the relevant application. However, the service was relying on old `mu-javascript-template` that was based off an unsupported node version (`v14.x.x`); both feature and security updates are no longer provided.

This PR bumps the template version in addition to a couple of packages inside `package.json`; it also updates the lock file.

## How to Test

1. Add `image: lblod/vendor-login-service:1.1.0-rc.1` to `docker-compose.override.yml`.

2. Insert a mock vendor in the database:

```sh
PREFIX muAccount: <http://mu.semte.ch/vocabularies/account/>
PREFIX mu:        <http://mu.semte.ch/vocabularies/core/>
PREFIX foaf:      <http://xmlns.com/foaf/0.1/>
PREFIX ext:       <http://mu.semte.ch/vocabularies/ext/>

INSERT DATA {
  GRAPH <http://mu.semte.ch/graphs/automatic-submission> {
    <http://example.com/vendor/mechelen> a foaf:Agent, ext:Vendor ;
      muAccount:key "mechelen-secret-key" ;
      muAccount:canActOnBehalfOf <http://data.lblod.info/id/bestuurseenheden/be278471a2a318edba32e7ac4294c0eafbe4c8077a34dcbb9c2e43211d4a78a6> ;
      foaf:name "Mechelen vendor" ;
      mu:uuid "aaaa-bbbb-cccc-1111-2222-3333" .
  }
}
```

3. Send the following cURL request:

```sh
curl -v -X POST \
      -H "Content-Type: application/json" \
      -b CookieJar.tsv -c CookieJar.tsv \
      -d '{
  "organization": "http://data.lblod.info/id/bestuurseenheden/be278471a2a318edba32e7ac4294c0eafbe4c8077a34dcbb9c2e43211d4a78a6",
  "publisher": {
    "uri": "http://example.com/vendor/mechelen",
    "key": "mechelen-secret-key"
  }
}' http://localhost:90/vendor/login
```

**NOTE**: The `:90` port is what the exposed identifier port in our development setups; it can be different depending on what you may have changed locally.

The request will save the cookie in a cookie jar (`CookieJar.tsv`), which will be used down the line to send SPARQL queries from a vendor. Check the **Response** subsection under [this section](https://github.com/lblod/vendor-login-service/blob/master/README.md#post-sessions) to compare your output and understand what it means; the expected code in this case should be `201 Created`.
